scapy/supersocket.py

```diff
@@ -106,8 +106,12 @@ def _recv_raw(self, sock, x):
                         pkt = pkt[:12] + tag + pkt[12:]
                 elif cmsg_lvl == socket.SOL_SOCKET and \
                         cmsg_type == SO_TIMESTAMPNS:
-                    tmp = struct.unpack("iiii", cmsg_data)
-                    timestamp = tmp[0] + tmp[2] * 1e-9
+                    length = len(cmsg_data)
+                    if length == 16:  # __kernel_timespec
+                        tmp = struct.unpack("ll", cmsg_data)
+                    elif length == 8:  # timespec
+                        tmp = struct.unpack("ii", cmsg_data)
```

Comment:
Can you add an `else:` case to handle an invalid length? That will prevent weird errors.